### ISSUE TITLE
Seeing occasional errors with my cron for pending sendouts

### ISSUE THREAD

**Issue by bossanova808** [putyourlightson/craft-campaign#143]
### Describe the bug
Seeing occasional errors with my cron for pending sendouts.

```
Cronic detected failure or error output for the command:
php /srv/users/serverpilot/apps/imagescience/craft/craft campaign/sendouts/run-pending-sendouts

RESULT CODE: 1

ERROR OUTPUT:
Exception 'yii\base\Exception' with message 'Has not waited the lock.'

in /srv/users/serverpilot/apps/imagescience/releases/dev/vendor/craftcms/cms/src/queue/Queue.php:562

Stack trace:
#0 /srv/users/serverpilot/apps/imagescience/releases/dev/vendor/craftcms/cms/src/queue/Queue.php(120): craft\queue\Queue->reserve()
#1 /srv/users/serverpilot/apps/imagescience/releases/dev/vendor/putyourlightson/craft-campaign/src/console/controllers/SendoutsController.php(37): craft\queue\Queue->run()
#2 [internal function]: putyourlightson\campaign\console\controllers\SendoutsController->actionRunPendingSendouts()
#3 /srv/users/serverpilot/apps/imagescience/releases/dev/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#4 /srv/users/serverpilot/apps/imagescience/releases/dev/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#5 /srv/users/serverpilot/apps/imagescience/releases/dev/vendor/yiisoft/yii2/console/Controller.php(164): yii\base\Controller->runAction('run-pending-sen...', Array)
#6 /srv/users/serverpilot/apps/imagescience/releases/dev/vendor/yiisoft/yii2/base/Module.php(528): yii\console\Controller->runAction('run-pending-sen...', Array)
#7 /srv/users/serverpilot/apps/imagescience/releases/dev/vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction('campaign/sendou...', Array)
#8 /srv/users/serverpilot/apps/imagescience/releases/dev/vendor/craftcms/cms/src/console/Application.php(87): yii\console\Application->runAction('campaign/sendou...', Array)
#9 /srv/users/serverpilot/apps/imagescience/releases/dev/vendor/yiisoft/yii2/console/Application.php(147): craft\console\Application->runAction('campaign/sendou...', Array)
#10 /srv/users/serverpilot/apps/imagescience/releases/dev/vendor/yiisoft/yii2/base/Application.php(386): yii\console\Application->handleRequest(Object(craft\console\Request))
#11 /srv/users/serverpilot/apps/imagescience/releases/dev/craft(22): yii\base\Application->run()
#12 {main}

```

Seems to happen only 'on the hours' - e.g. 6pm and 12am.  I am thinking something else is being queued at the same time, but I am not getting other errors from anything else as far as I can see.

### Versions
- Plugin version:  1.15.3
- Craft version: 3.4.19.1

**Comment by bencroker** [comment]
This is a Craft exception that is thrown because the queue is already being run, so there is nothing to worry about. If you have other cron jobs also running at regular intervals then you could perhaps space them out so that they don't run at the same time.

**Comment by bossanova808** [comment]
I'm surprised just as a general thing, Craft doesn't just wait on the lock but rather raises an exception. 

 I've got a bunch of things running regularly, and they all tend to use e.g.

```
*/10 * * * * cronic php /srv/users/serverpilot/apps/imagescience/craft/craft campaign/sendouts/run-pending-sendouts
```
That `*/10` means inevitably the same times I guess (I have some `*/12` and `*/15`s for example) - they will all land on the hour, though.

I tried `5/10` which I read should make things run a 5,15,25... etc but crontab got grumpy, so I have ended up at 

```
5,15,25,35,45,55 * * * * cronic php /srv/users/serverpilot/apps/imagescience/craft/craft campaign/sendouts/run-pending-sendouts
```
...which cron likes and should stop those things conflicting.

Thanks again!
